lib/ASTGen/Sources/ASTGen/PluginHost.swift

```diff
@@ -349,6 +349,10 @@ class PluginDiagnosticsEngine {
     guard let bufferBaseAddress = exportedSourceFile.pointee.buffer.baseAddress else {
       return nil
     }
+    // Ensure 'offset' is within the buffer.
+    guard offset <= exportedSourceFile.pointee.buffer.count else {
```

Comment:
`<=` because the end of the  buffer is part of the buffer.
https://github.com/apple/swift/blob/9f067c19405f3380e5206fe95ea8e878abd04c11/lib/Basic/SourceLoc.cpp#L498-L500